src/LinuxTracing/PerfEventReaders.cpp

```diff
@@ -494,6 +494,23 @@ void ReadPerfSampleIdAll(PerfEventRingBuffer* ring_buffer, const perf_event_head
   std::memcpy(&sched_wakeup, res.raw_data.get(), sizeof(sched_wakeup_tracepoint_fixed));
 
   ring_buffer->SkipRecord(header);
+
+  // If we did not received the necessary data for a callstack, there is no need to return a
+  // SchedSwitchWithStackPerfEvent.
+  if (res.dyn_size == 0 || res.stack_data == nullptr || res.regs == nullptr) {
+    return SchedWakeupPerfEvent{
+        .timestamp = res.time,
+        .ordered_stream = PerfEventOrderedStream::FileDescriptor(ring_buffer->GetFileDescriptor()),
+        .data =
+            {
+                // The tracepoint format calls the woken tid "data.pid" but it's effectively the
+                // thread id.
```

Comment:
I would avoid repeating this comment. It's just maintenance to keep them in sync. Consider a small comment that simply points to the other one.

Reply:
Done.